packages/amplify-migration-e2e/src/templategen.ts

```diff
@@ -0,0 +1,175 @@
+  const pattern = /```([\s\S]*?)```/g;
+  const matches = readmeContent.matchAll(pattern);
+  const commands = [];
+
+  for (const match of matches) {
+    if (match[1]) {
+      commands.push(match[1].trim());
+    }
+  }
+  if (commands.length === 0) {
+    throw new Error('README file parsing failed to get the stack refactor commands');
+  }
+  return commands;
+}
+
+function getCommandsFromReadme(readmeContent: string) {
+  const step1Content = extractContent(readmeContent, '### STEP 1', '#### Rollback step');
+  const step2Content = extractContent(readmeContent, '### STEP 2', '#### Rollback step');
+  const step3Content = extractContent(readmeContent, '### STEP 3', '#### Rollback step');
+  const step1Commands = extractCommands(step1Content);
+  const step2commands = extractCommands(step2Content);
+  const step3Commands = extractCommands(step3Content);
+  return { step1Commands, step2commands, step3Commands };
+}
+
+async function executeCommand(command: string, cwd?: string) {
+  cwd = cwd ?? process.cwd();
+  const processResult = execa.sync(command, {
+    cwd,
+    env: { ...process.env, npm_config_user_agent: 'npm' },
+    encoding: 'utf-8',
+    shell: true,
+  });
+
+  if (processResult.exitCode === 0) {
+    return processResult.stdout;
+  } else {
+    throw new Error(`Command exit code: ${processResult.exitCode}, message: ${processResult.stderr}`);
+  }
+}
+
+async function executeCreateStackRefactorCallCommand(command: string, cwd: string) {
+  const processResult = JSON.parse(await executeCommand(command, cwd));
+  const stackRefactorId = processResult.StackRefactorId;
+  return stackRefactorId;
+}
+
+async function assertStepCompletion(command: string) {
+  const processResult = JSON.parse(await executeCommand(command));
+  return processResult.Stacks[0].StackStatus;
+}
+
+async function assertRefactorStepCompletion(command: string) {
+  const processResult = JSON.parse(await executeCommand(command));
+  return processResult.Status;
+}
+
+async function executeStep1(cwd: string, commands: string[]) {
+  await executeCommand(commands[0], cwd);
+  await retry(
+    () => assertStepCompletion(commands[1]),
+    (status) => status.includes('COMPLETE') && !status.includes('IN_PROGRESS'),
+    {
+      times: 50,
+      delayMS: 1000, // 1 second
+      timeoutMS: 1000 * 60 * 5, // 5 minutes
+    },
+    (status) => status.includes('FAILED'),
+  );
+}
+
+async function executeStep2(cwd: string, commands: string[]) {
+  await executeCommand(commands[0], cwd);
+  await retry(
+    () => assertStepCompletion(commands[1]),
+    (status) => status.includes('COMPLETE') && !status.includes('IN_PROGRESS'),
+    {
+      times: 50,
+      delayMS: 1000, // 1 second
+      timeoutMS: 1000 * 60 * 5, // 5 minutes
+    },
+    (status) => status.includes('FAILED'),
+  );
+}
+
+async function executeStep3(cwd: string, commands: string[], bucketName: string) {
+  setEnvVariable('BUCKET_NAME', bucketName);
+  await executeCommand(commands[1], cwd);
```

Comment:
can we name the commands instead of accessing by indices for easier understanding: e.g. `const [uploadSourceTemplate , uploadDestinationTemplate, ...]` ?

Reply:
If we follow the Black Box Testing approach, this approach is irrelevant then?

Reply:
Yup!